Cargo.lock

```diff
@@ -2293,15 +2290,16 @@ dependencies = [
  "once_cell 1.3.1 (registry+https://github.com/rust-lang/crates.io-index)",
  "proptest 0.9.6 (registry+https://github.com/rust-lang/crates.io-index)",
  "proptest-derive 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
- "rand 0.6.5 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rand 0.7.3 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rand_core 0.5.1 (registry+https://github.com/rust-lang/crates.io-index)",
```

Comment:
Why do we need this?

Reply:
To reflect the change in dependency operated by dalek, which no longer uses the `rand` trait bounds, replacing them by those of `rand_core`:
https://github.com/dalek-cryptography/curve25519-dalek/pull/298

We now use those same `rand_core`-based traits in places where we e.g. generate random keys rather than overlaying them with a sufficient-but-not-necessary `rand` dependency and introducing another source of incompatibility. 

Besides, `rand` is still used in tests and executables, to actually instantiate the (thread—, Std—)Rngs that yield our actual keys and configs.

Doing things this way insulates us better from changes in `rand`. Seeing as you're looking at a Cargo lock, the added rand_core dependency line in one version or another was unavoidable (see https://github.com/rust-random/rand/issues/648 for details).